### PR TITLE
News

### LINKED ISSUE #7: news feed
list of news articles found on https://www.elixirsolutions.com/news. drive from the query index. probably can get rid of the paging stuff and just use an infinite scroll/show more approach, but should check with customer on that.

May want to make this and #8 1 block just with variants or author configurable filters

### LINKED ISSUE #24: news page theme
News pages have a theme (class on the body) of news applied in metadata.xlsx

need to format them into 2 cols 2/3 1/3 layout based on the is theme

### PR DESCRIPTION
**Pull request by shsteimer**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #7 
Fix #24 

Test URLs:
- Before: https://main--elixirsolutions--hlxsites.hlx.live/
- After: https://news--elixirsolutions--hlxsites.hlx.live/news
- After: https://news--elixirsolutions--hlxsites.hlx.live/news/tribal-edge

### CODE REVIEW

File: blocks/news-feed/news-feed.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  `;
+
+  return postCard;
+}
+
+async function buildNewsFeed(ul, pageNum, pageControl) {
+  const limit = 5;
+  const offset = pageNum * limit;
+  let morePages = false;
+  const blogPosts = ffetch('/query-index.json')
```
Comment (ddragosd) [inline blocks/news-feed/news-feed.js:32]: I'd rename `query-index` to something like `news-feed.json` and also leave a comment in the code that its content is maintained in `news-feed.xlsx` - whoever ends up maintaining this site on RiteAid and doesn't know Hlx that well, to find it more manageable.

Comment (ddragosd) [reply blocks/news-feed/news-feed.js:32]: Also if the spreadsheet is to be maintained by authors, can we change the columns `lastModified` and `date` into a Date cell format in excel ?

Comment (ddragosd) [reply blocks/news-feed/news-feed.js:32]: I'd also rename the var `blogPosts` into `newsArticles`

File: blocks/news-feed/news-feed.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    const li = document.createElement('li');
+    li.append(buildPost(post, i < 1));
+    newUl.append(li);
+
+    i += 1;
+  }
+
+  pageControl.innerHTML = `
+      <ul class="pages">
+        <li class="prev"><a data-page="${pageNum - 1}" href="${window.location.pathname}?page=${pageNum}"><span class="icon icon-next"><span class="sr-only">Previous Page</span></a></li>
```
Comment (ddragosd) [inline blocks/news-feed/news-feed.js:56]: I like that `data-page` is maintained here. Documenting this method to specify how `pageNum` is maintained would be great.

File: blocks/news-feed/news-feed.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          <span>Read More</span>
+        </a>
+      </div>
+    </a>
+  `;
+
+  return postCard;
+}
+
+async function buildNewsFeed(ul, pageNum, pageControl) {
```
Comment (ddragosd) [inline blocks/news-feed/news-feed.js:28]: `pageControl` is an HTML Element. For clarity I'd rename it into `pageContent` (initially I thought it's a controller)

File: scripts/scripts.js [hunk 4]
```diff
@@ -16,11 +16,44 @@ import {
...
+
+  const h1 = main.querySelector('h1');
+  if (!h1) {
+    return;
+  }
+  const section = h1.closest('div');
+  const firstColElems = [];
+  const secondColElems = [];
+  let h1found = false;
+  [...section.children].forEach((elem) => {
```
Comment (ddragosd) [inline scripts/scripts.js:32]: I'd try to use `.filter` to make the code easier to read:
```
const firstColElems = [...section.children].filter(elem => elem === h1);
```

File: blocks/news-feed/news-feed.js (hunk 1 above)
Comment (shsteimer) [reply blocks/news-feed/news-feed.js:32]: `query-index` is a standard name across all projects, so I'm disinclines to rename that. the content is not author maintined, see https://www.hlx.live/developer/indexing


renaming the variable is a good idea, will make that change

File: scripts/scripts.js (hunk 4 above)
Comment (shsteimer) [reply scripts/scripts.js:32]: I don't think that will work.

the point of this code is that any element before the H1 goes in the first col, and anything after goes in the second col. so we are not simply finding the h1, rather, we are looking all the elements, and segmenting the into 2 arrays based on the h1 position.

File: blocks/news-feed/news-feed.js (hunk 2 above)
Comment (shsteimer) [reply blocks/news-feed/news-feed.js:60]: added a comment which hopefully clarifies that

**Pull request merged**